File: python_cpp/uPIMulator_frontend/src/abi/isa/cc/div_cc.py
from typing import Set

from abi.isa.cc._base_cc import BaseCC
from abi.isa.instruction.condition import Condition


class DivCC(BaseCC):
    def __init__(self, condition: Condition):
        super().__init__(
            DivCC.conditions(), condition,
        )

    @staticmethod
    def conditions() -> Set[Condition]:
        return {
            Condition.SZ,
            Condition.SNZ,
            Condition.SPL,
            Condition.SMI,
            Condition.TRUE,
            Condition.FALSE,
        }

File: golang/uPIMulator/src/linker/kernel/instruction/cc/div_cc.go
package cc

import (
	"errors"
)

type DivCc struct {
	condition Condition
}

func (this *DivCc) Init(condition Condition) {
	conditions := map[Condition]bool{
		SZ:    true,
		SNZ:   true,
		SPL:   true,
		SMI:   true,
		TRUE:  true,
		FALSE: true,
	}

	if _, found := conditions[condition]; !found {
		err := errors.New("condition is not allowed")
		panic(err)
	}

	this.condition = condition
}

func (this *DivCc) Condition() Condition {
	return this.condition
}
